### DISCUSSION TITLE
SchemaSpy would be easier to use if?

### DISCUSSION THREAD

**Discussion by npetzall** [schemaspy/schemaspy#990] [Q&A]
When you've used it for as long as I have. You think everything is obvious and expected.

What could be done to make SchemaSpy easier to use?

Documentation is good, but better when it's not required.

**Comment by jjjasper** [comment]
the number one thing that would make things easier for my company would be a release! There are issues that have been fixed but not available in a release.

**Comment by toraritte** [comment]
As a first time user, I was frustrated with the command line help, the inconsistent flag styles (`-`, `--`), trying to figure out how to call which (e.g., neither `--user <user>` nor `--user=<user>` worked, only with `-u`), and some were not even documented, such as (`-p` in `schemaspy --help`, that is). For the record, this was all in 6.1.1.

With that said, after figuring out the quirks, I was blown away by the generated output. I just inherited a fairly complex web service with no documentation, and thus I'm also transitioning from casual PostgreSQL user to full-blown database admin, and I have to say  that SchemaSpy is simply amazing. I found it by accident, and some larger discussions (e.g., on Stackoverflow) don't even mention it. I especially love the sub-graphs relative to a table. I'm mostly using `psql` and **pgAdmin 4** for certain tasks, but the SchemaSpy gave me superpowers to get an overview on the system (and also an instant documentation). **pgAdmin 4**  has an ERD tool, but it is way less useful (especially because most of the time it gets stuck loading).

Thank you for all of your time and effort for this awesome tool!

**edit**: Just looked up the [open issues for the `documentation` tag](https://github.com/schemaspy/schemaspy/issues?q=is%3Aissue+is%3Aopen+documentation+label%3Adocumentation), and hope I'll be able to contribute soon!

**Reply by npetzall** [reply]
@toraritte Hi, thanks for your kind words.

We did a big refactoring for 6.2 just to solve the command line issues, we had gotten stuck in a semi transitioned state so we finished that up before releasing 6.2, we still have work to do with the help. Now it shows all options and...you go from to few to too many.

If you download the latest 6.2.2 and use `-help` you'll see what I mean.
